.github/workflows/proxy_plugin.yml

```diff
@@ -0,0 +1,74 @@
+---
+name: Foreman Proxy Plugin
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        description: Plugin name
+        required: true
+        type: string
+      foreman_proxy_version:
+        description: Foraman proxy version to use
+        default: 'develop'
+        required: false
+        type: string
+      foreman_proxy_repository:
+        description: Foreman proxy repository to use
+        default: 'theforeman/smart-proxy'
+        required: false
+        type: string
+      matrix_include:
+        description: Additional test matrix combinations to include
+        default: '[]'
+        required: false
+        type: string
+      matrix_exclude:
+        description: Test matrix combinations to exclude
+        default: '[]'
+        required: false
+        type: string
+
+env:
+  BUNDLE_WITHOUT: "bmc:development:dhcp_isc_inotify:dhcp_isc_kqueue:journald:krb5:libvirt:puppetca_token_whitelisting:realm_freeipa:windows"
+
+jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: theforeman/gha-matrix-builder@v0
+        with:
+          repository: ${{ inputs.foreman_proxy_repository }}
+          ref: ${{ inputs.foreman_proxy_version }}
+          include: ${{ inputs.matrix_include }}
+          exclude: ${{ inputs.matrix_exclude }}
+
+  test:
+    name: "Foreman Proxy ${{ inputs.foreman_proxy_version }} with Ruby ${{ matrix.ruby }}"
```

Comment:
this is a lie, we take the matrix from that version, but we never setup that proxy ourself

Reply:
We now do, as we pass `SMART_PROXY_BRANCH` env var, which will carry the right branch to check out in Gemfile of the plugin